### PR TITLE
[update] Project config and some docs

### PR DESCRIPTION
**Pull request by nhunzaker**
New project settings. I discovered some neat things:

1. [Babel can inline environment variables](http://babeljs.io/docs/usage/transformers/utility/inline-environment-variables/)
2. Makefiles are awesome
3. I added `private: true` to the package.json so that it can not be published through `npm publish`. `npm run release` or `make release` disables this, allowing the proper release process.
4. Karma has an `autoWatchBatchDelay` setting, which seems to address issues where `karma-webpack` causes double test runs.

### Why a makefile?

I found myself pulling out a bunch of commands into shell scripts and very long `script` declarations in the package.json. Make lets me keep this all in a well organized place. Also, I like how easy it makes it to call other tasks and generate lists of dependencies.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,12 @@
 {
   "stage": 1,
-  "loose": ["es6.modules", "es6.classes"],
-  "sourceMaps" : "inline"
+  "loose": true,
+  "sourceMaps": "inline",
+  "optional"  : [
+    "minification.propertyLiterals",
+    "minification.memberExpressionLiterals",
+    "utility.inlineEnvironmentVariables",
```
Comment (nhunzaker) [inline .babelrc]: Babel can do environment variables for you!

File: karma.conf.js [hunk 2]
```diff
@@ -1,21 +1,13 @@
...
-
   config.set({
-
-    browsers: browsers,
+    browsers: [ 'Chrome' ],
 
     frameworks: [ 'mocha', 'sinon-chai' ],
 
-    logLevel: config.LOG_ERROR,
+    autoWatchBatchDelay: 400,
```
Comment (nhunzaker) [inline karma.conf.js]: This helps to address cases where karma double runs

File: karma.conf.js [hunk 3]
```diff
@@ -37,28 +29,15 @@ module.exports = function(config) {
...
-          {
-            test    : /\.json$/,
-            loader  : 'json'
-          }
-        ],
+        loaders: [{
+          test    : /\.jsx*$/,
+          exclude : /node_modules/,
+          loader  : 'babel',
+          query   : { optional: ['runtime'] }
```
Comment (nhunzaker) [inline karma.conf.js]: Other options are not necessary because of  `.babelrc`

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+javascript-min: dist/src/Microcosm.js
+	@$(WEBPACK) -p $^ dist/microcosm.build.js \
+	--devtool sourcemap --output-library-target commonjs2
+
+javascript: $(shell find src -name '*.js' ! -path '*/__tests__/*')
+	@mkdir -p dist
+	@$(BABEL) -d dist $^
+	@make javascript-min
+
```
Comment (nhunzaker) [inline Makefile]: `$^` in here says "Take the list of files described on line 19 and spread them out here"

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,52 @@
+BIN     = $$(npm bin)
+BABEL   = $(BIN)/babel
+KARMA   = $(BIN)/karma
+WEBPACK = $(BIN)/webpack
+
+.PHONY: clean test test-watch build package.json javascript release example javascript-min
+
+build:
+	@make clean
```
Comment (nhunzaker) [inline Makefile]: `@` before a command silences it, I'm using this to give cleaner build generation logging

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,52 @@
+BIN     = $$(npm bin)
+BABEL   = $(BIN)/babel
+KARMA   = $(BIN)/karma
+WEBPACK = $(BIN)/webpack
+
+.PHONY: clean test test-watch build package.json javascript release example javascript-min
```
Comment (nhunzaker) [inline Makefile]: .PHONY tells make to _always_ run a command even if the files listed (see the `javascript` task) have not updated. For now, I'm more or less phonying everything. I'd like to be smarter about this.

File: .babelrc (hunk 1 above)
Comment (nhunzaker) [inline .babelrc]: This means it will automatically replace things like `process.env.NODE_ENV` with `"development"`

File: karma.conf.js (hunk 2 above)
Comment (nhunzaker) [inline karma.conf.js]: @greypants: This _might_ address the issue you were bumping into with the build compiling twice. It's working for me, at any rate.

**Comment by solomonhawk** [comment]
I must learn more about Make. :+1:

**Comment by nhunzaker** [comment]
@solomonhawk I'm mixed about it. I have loved working with it, but it is definitely alien at firstis alien.

**Pull request merged**